pulseapi/entries/views.py

```diff
@@ -4,13 +4,54 @@
 
 import django_filters
 from rest_framework import (filters, status)
-from rest_framework.decorators import detail_route
+from rest_framework.decorators import detail_route, api_view
 from rest_framework.generics import ListCreateAPIView, RetrieveAPIView
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.response import Response
 
 from pulseapi.entries.models import Entry
 from pulseapi.entries.serializers import EntrySerializer
+from pulseapi.users.models import UserFavorites
+
+# FIXME: TODO: Remove the handling for GET, so that only PUT works
+@api_view(['GET','PUT'])
+def toggle_favorite(request, entryid):
```

Comment:
line 17 says "only expose this `def` when the http verb is GET or PUT (http verb whitelist)

line 18 is a function that services a view (see urls.py) as a general function rather than a class entry, as faving an entry is technically something "outside" of entries and users.